examples/plot_incr_kmeans.py

```diff
@@ -0,0 +1,319 @@
+
+
+
+# Active learning like-experiment
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# One core element of the experiment differs from active learning. In fact,
+# in active learning, we do not expect to select a number of samples equal to
+# the number of real cluster in the data. We also build our clustering over
+# several iterations. We now try to do the same on generated data.
+
+X, y, centers = make_blobs(n_samples=10000, centers=100, return_centers=True,
+                           n_features=512, random_state=2)
+
+clf = RandomForestClassifier()
+
+kmeans_inertia = []
+kmeans_accuracy = []
+sampler = MiniBatchKMeansSampler(10)
+
+idx = ActiveLearningSplitter(10000, test_size=.2, random_state=0)
+idx.add_batch(np.arange(10))
+
+for i in range(10):
+    selected = sampler.fit(X[idx.selected]).select_samples(X[idx.non_selected])
+    idx.add_batch(selected)
+    kmeans_inertia.append(inertia(X[idx.test], X[idx.selected]))
+
+    clf.fit(X[idx.selected], y[idx.selected])
+    kmeans_accuracy.append(clf.score(X[idx.test], y[idx.test]))
+    
+
+ikmeans_inertia = []
+ikmeans_accuracy = []
+sampler = IncrementalMiniBatchKMeansSampler(10, random_state=0)
+
+idx = ActiveLearningSplitter(10000, test_size=.2, random_state=0)
+idx.add_batch(np.arange(10))
+
+for i in range(10):
+
+    selected = sampler.fit(X[idx.selected]).select_samples(X[idx.non_selected], fixed_cluster_centers=X[idx.selected])
+
+    idx.add_batch(selected)
+    ikmeans_inertia.append(inertia(X[idx.test], X[idx.selected]))
+
+    clf.fit(X[idx.selected], y[idx.selected])
+    ikmeans_accuracy.append(clf.score(X[idx.test], y[idx.test]))
+
+
+
+irkmeans_inertia = []
+irkmeans_accuracy = []
+sampler = IncrementalMiniBatchKMeansSampler(10, random_state=0)
+
+idx = ActiveLearningSplitter(10000, test_size=.2, random_state=0)
+idx.add_batch(np.arange(10))
+
+for i in range(10):
+
+    selected = sampler.fit(X[idx.selected]).select_samples(X[idx.non_selected], fixed_cluster_centers=X[idx.selected], recenter_every=100)
+
+    idx.add_batch(selected)
+    irkmeans_inertia.append(inertia(X[idx.test], X[idx.selected]))
+
+    clf.fit(X[idx.selected], y[idx.selected])
+    irkmeans_accuracy.append(clf.score(X[idx.test], y[idx.test]))
+
+plt.plot(range(10), kmeans_accuracy, label='KMeans')
+plt.plot(range(10), ikmeans_accuracy, label='Incr. KMeans')
+plt.plot(range(10), irkmeans_accuracy, label='Incr. KMeans R')
+plt.ylabel('Accuracy ────')
+plt.legend(loc=6)
+plt.xlabel('Iteration')
+
+plt.gca().twinx()
+plt.gca().set_prop_cycle(None)
+plt.plot(range(10), kmeans_inertia, '--')
+plt.plot(range(10), ikmeans_inertia, '--')
+plt.plot(range(10), irkmeans_inertia, '--')
+plt.ylabel('Inertia ╶╶╶╶')
+
+plt.tight_layout(rect=[0, 0.03, 1, 0.95])
+plt.show()
+
+##############################################################################
+# In this simple exemple, we see that Incremental KMeans has a decisive
+# advantage on KMeans because it is able to gradually explore the space of
```

Comment:
```suggestion
# advantage over KMeans because it is able to gradually explore the space of
```
```suggestion
# advantage on KMeans because it is able to gradually explore the space of
```